### PR TITLE
Adding TM+AGW template

### PR DESCRIPTION
**Pull request by AlekseiPolkovnikov**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
*
*
*
*
*

### CODE REVIEW

File: traffic-manager-application-gateway-demo-setup/azuredeployappgw.json [hunk 1]
```diff
@@ -0,0 +1,342 @@
+﻿{
```
Comment (bmoore-msft) [inline traffic-manager-application-gateway-demo-setup/azuredeployappgw.json]: this should be in a subfolder - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#nested-templates

File: traffic-manager-application-gateway-demo-setup/azuredeployappgw.json [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    "ipId": {
+      "type": "string",
+      "value": "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPAddressName'))]"
+    }
+  },
+  "resources": [
+    {
+      "apiVersion": "2015-01-01",
+      "name": "[variables('wsDeploymentName1')]",
+      "type": "Microsoft.Resources/deployments",
```
Comment (bmoore-msft) [inline traffic-manager-application-gateway-demo-setup/azuredeployappgw.json]: copy loop?

File: traffic-manager-application-gateway-demo-setup/azuredeployappgw.json [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+    "appGatewayIPConfigName": "[concat(parameters('appGwName'),'-ipconfig')]",
+    "appGatewayrequestRoutingRuleName": "[concat(parameters('appGwName'),'-rule1')]",
+    "appGatewayHttpListenerName": "[concat(parameters('appGwName'),'-listener')]",
+    "appGatewayBackendHttpSettingsName": "[concat(parameters('appGwName'),'-httpsettings')]",
+    "appGatewayBackendPoolName": "[concat(parameters('appGwName'),'-backendpool')]",
+    "appGatewayFrontendPortName": "[concat(parameters('appGwName'),'-port')]",
+    "appGatewayFrontendIPConfigName": "[concat(parameters('appGwName'),'-ipconfig')]",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetPrefix": "10.0.0.0/28",
+    "webServerTemplateLocation": "https://raw.githubusercontent.com/AlekseiPolkovnikov/azure-quickstart-templates/tmagw/traffic-manager-application-gateway-demo-setup/",
```
Comment (bmoore-msft) [inline traffic-manager-application-gateway-demo-setup/azuredeployappgw.json]: see BP comment above

File: traffic-manager-application-gateway-demo-setup/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+          },
+          "appGwName": {
+            "value": "appgw1"
+          }
+        }
+      }
+    },
+    {
+      "apiVersion": "2015-01-01",
+      "name": "appGw2",
```
Comment (AlekseiPolkovnikov) [inline traffic-manager-application-gateway-demo-setup/azuredeploy.json]: This was considered, however, having only 2 instances, input that is parameter names bound (parameters(primarylocation'))/parameters('secondarylocation')), and deployments output handling, looked much easier?

File: traffic-manager-application-gateway-demo-setup/azuredeployappgw.json (hunk 2 above)
Comment (AlekseiPolkovnikov) [inline traffic-manager-application-gateway-demo-setup/azuredeployappgw.json]: Same as above. Do we consider the overhead worthwhile for 2 instances? How about return params?

**Comment by AlekseiPolkovnikov** [comment]
@bmoore-msft thanks, apart from copy loops (see replies above), fixed. Please let me know if this is sufficient, will need to apply same changes to https://github.com/Azure/azure-quickstart-templates/pull/2490

**Comment by bmoore-msft** [comment]
re: copy loop - I think this would work well, you can create an array variable that holds your locations - similar on the output side.  Just output and array and consume accordingly.  The "overhead" of the array is much less than duplicating all the code needed for the deployment.

re: other changes - you have two nested folders, i.e. nested/nested is that intentional necessary?

All the other changes look good...

**Comment by AlekseiPolkovnikov** [comment]
@bmoore-msft  thanks for the explanation. My point was it kind of loses expressiveness with too much inside "parametrization" in case of just two instances.. 
Anyway, please have a look at modifications requested. 
Regarding: nested - yes, there are 2 levels of nesting in this template, so it is reflected in folder structure.

**Comment by AlekseiPolkovnikov** [comment]
@bmoore-msft same changes introduced to https://github.com/Azure/azure-quickstart-templates/pull/2490

File: traffic-manager-application-gateway-demo-setup/azuredeploy.json [hunk 5]
```diff
@@ -96,43 +96,20 @@
     "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
     "addressPrefix": "10.0.0.0/16",
     "subnetPrefix": "10.0.0.0/28",
-    "nestedTemplatesFolder":  "nested",
-    "appGwTemplateName": "azuredeployappgw.json"
+    "nestedTemplatesFolder": "nested",
+    "appGwTemplateName": "azuredeployappgw.json",
+    "location1": "[parameters('primarylocation')]",
+    "location2": "[parameters('secondarylocation')]"
```
Comment (bmoore-msft) [inline traffic-manager-application-gateway-demo-setup/azuredeploy.json]: Maybe use: 
    "location": [
      "[parameters('primaryLocation')]", 
      "[parameters('secondaryLocation')]"
    ]
And then reference via:
variables('location')[copyIndex()]

**Pull request merged**